virtio-devices/src/balloon.rs

```diff
@@ -138,6 +179,49 @@ impl BalloonEpollHandler {
     }
 
     fn release_memory_range(
+        &mut self,
+        memory: &GuestMemoryMmap,
+        range_base: GuestAddress,
+        range_len: usize,
+    ) -> result::Result<(), Error> {
+        let page_size: u64 = get_page_size();
+        if page_size == 1 << VIRTIO_BALLOON_PFN_SHIFT {
+            return Self::discard_memory_region(memory, range_base, range_len);
+        }
+
+        self.pbp = Some(PartiallyBalloonedPage::new());
```

Comment:
This is not right. You only need to new the `pbp` if it is not initialized (e.g. self.pbp is none).

Reply:
Yes, my bad.